plugins/feeds/public/et_open.py

```diff
@@ -0,0 +1,138 @@
+import datetime
+import logging
+from datetime import timedelta, timezone
+from io import StringIO
+
+from idstools import rule
+
+from core import taskmanager
+from core.config.config import yeti_config
+from core.schemas import entity, indicator, task
+
+
+class ETOpen(task.FeedTask):
+    __SOURCE = (
+        "https://rules.emergingthreats.net/open/suricata-7.0.3/emerging-all.rules"
+    )
+
+    _defaults = {
+        "frequency": timedelta(days=1),
+        "name": "ETOpen",
+        "description": "ETOpen ruleset",
+        "source": __SOURCE,
+    }
+
+    def run(self):
+        response = self._make_request(self.__SOURCE, no_cache=True)
+
+        if not response:
+            return
+
+        for line in StringIO(response.text).readlines():
+            if line.startswith("#"):
+                continue
+            self.analyze(line)
+
+    def analyze(self, line: str):
+        rule_suricata = rule.parse(line)
+        if not rule_suricata:
+            return
+
+        if self.__filter_rule(rule_suricata.metadata):
+            ind_suricator = indicator.Suricata(
+                name=rule_suricata["msg"],
+                pattern=line,
+                metadata=rule_suricata.metadata,
+                diamond=indicator.DiamondModel.infrastructure,
+                sid=rule_suricata["sid"],
+            ).save()
+
+            if "cve" in ",".join(rule_suricata.metadata):
+                for ind_cve in self.__extract_cve(rule_suricata.metadata):
+                    ind_suricator.link_to(ind_cve, "affect", "ETOpen")
+
+            if "malware family" in ",".join(rule_suricata.metadata):
+                for in_malware_family in self.__extract_malware_family(
+                    rule_suricata.metadata
+                ):
+                    ind_suricator.link_to(in_malware_family, "affect", "ETOpen")
+
+            tags = self.__extract_tags(rule_suricata.metadata)
+            if tags:
+                ind_suricator.tag(tags)
+            if "mitre_tactic_id" in ",".join(rule_suricata.metadata):
+                for ind_mitre_attack in self.__extract_mitre_attack(
+                    rule_suricata.metadata
+                ):
+                    if ind_mitre_attack:
+                        ind_suricator.link_to(ind_mitre_attack, "affect", "ETOpen")
+
+    def __extract_cve(self, metadata: list):
```

Comment:
Use stronger typing here:

```suggestion
    def __extract_cve(self, metadata: list[str]) -> entity.Vulnerability:
```

Also do this in other functions